src/test/ui/const-generics/defaults/intermixed-correct-error.stderr

```diff
@@ -0,0 +1,280 @@
+
+error[E0107]: wrong number of type arguments: expected at most 1, found 2
+  --> $DIR/intermixed-correct-error.rs:44:17
+   |
+LL | type ConstTy3 = Example<f32, f32, 3>;
+   |                 ^^^^^^^^^^^^^^^^^^^^ expected at most 1 type argument
+
+error[E0107]: wrong number of lifetime arguments: expected 1, found 2
+  --> $DIR/intermixed-correct-error.rs:48:28
+   |
+LL | type ConstLt1 = Example<3, 'static, 'static>;
+   |                            ^^^^^^^ unexpected lifetime argument
+
+error[E0107]: wrong number of lifetime arguments: expected 1, found 2
+  --> $DIR/intermixed-correct-error.rs:50:37
+   |
+LL | type ConstLt2 = Example<'static, 3, 'static>;
+   |                                     ^^^^^^^ unexpected lifetime argument
+
+error[E0107]: wrong number of lifetime arguments: expected 1, found 2
+  --> $DIR/intermixed-correct-error.rs:52:34
+   |
+LL | type ConstLt3 = Example<'static, 'static, 3>;
+   |                                  ^^^^^^^ unexpected lifetime argument
+
+error[E0107]: wrong number of lifetime arguments: expected 1, found 2
+  --> $DIR/intermixed-correct-error.rs:55:27
+   |
+LL | type TyLt1 = Example<f32, 'static, 'static>;
+   |                           ^^^^^^^ unexpected lifetime argument
+
+error[E0107]: wrong number of const arguments: expected 1, found 0
+  --> $DIR/intermixed-correct-error.rs:55:14
+   |
+LL | type TyLt1 = Example<f32, 'static, 'static>;
+   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 1 const argument
+
+error[E0107]: wrong number of lifetime arguments: expected 1, found 2
+  --> $DIR/intermixed-correct-error.rs:58:36
+   |
+LL | type TyLt2 = Example<'static, f32, 'static>;
+   |                                    ^^^^^^^ unexpected lifetime argument
+
+error[E0107]: wrong number of const arguments: expected 1, found 0
+  --> $DIR/intermixed-correct-error.rs:58:14
+   |
+LL | type TyLt2 = Example<'static, f32, 'static>;
+   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 1 const argument
+
+error[E0107]: wrong number of lifetime arguments: expected 1, found 2
+  --> $DIR/intermixed-correct-error.rs:61:31
+   |
+LL | type TyLt3 = Example<'static, 'static, f32>;
+   |                               ^^^^^^^ unexpected lifetime argument
+
+error[E0107]: wrong number of const arguments: expected 1, found 0
+  --> $DIR/intermixed-correct-error.rs:61:14
+   |
+LL | type TyLt3 = Example<'static, 'static, f32>;
+   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 1 const argument
+
+error[E0107]: wrong number of const arguments: expected 1, found 2
+  --> $DIR/intermixed-correct-error.rs:65:30
+   |
+LL | type TyConst1 = Example<f32, 3, 3>;
+   |                              ^ unexpected const argument
+
+error[E0107]: wrong number of const arguments: expected 1, found 2
+  --> $DIR/intermixed-correct-error.rs:68:33
+   |
+LL | type TyConst2 = Example<3, f32, 3>;
+   |                                 ^ unexpected const argument
+
```

Comment:
This error is less nice than I thought :thinking: 

It would probably a good idea to match each generic param with the expected generics after dealing with elided parameters.

@JulianKnodt do you want to try this yourself?

Reply:
Ah sure, what specifically do you mean though?

Reply:
modify the current point in the pipeline and instead of comparing counts I think you pretty much first check lifetimes and then walk the expected and actual generics in parallel mentioning each mismatch in the final error. Potentially even being clever for cases like `<7, 7, u32>` when `<const: N: usize, T>` is expected by eating the second const arg, not sure how easy that is though.

Reply:
Would this sort of loop be alright?